tests/Admin/ArticleAdminTest.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace Tests\AppBundle\Admin;
+
+use AppBundle\DataFixtures\ORM\LoadAdminData;
+use AppBundle\DataFixtures\ORM\LoadArticleData;
+use AppBundle\Entity\Article;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Tests\AppBundle\Controller\ControllerTestTrait;
+use Tests\AppBundle\MysqlWebTestCase;
+
+/**
+ * @group functional
+ */
+class ArticleAdminTest extends MysqlWebTestCase
+{
+    use ControllerTestTrait;
+
+    public function testCreateCategoryFail(): void
+    {
+        $this->authenticateAsAdmin($this->client, 'superadmin@en-marche-dev.fr', 'superadmin');
+
+        $crawler = $this->client->request(Request::METHOD_GET, '/admin/app/article/create');
+        $this->assertStatusCode(Response::HTTP_OK, $this->client);
+        $form = $crawler->selectButton('btn_create_and_edit')->form();
+        $this->client->submit($form);
+
+        $this->assertStatusCode(Response::HTTP_OK, $this->client);
+        $this->assertValidationErrors(
+            [
+                'data.title',
+                'data.description',
+                'data.content',
+                'data.media',
+                'data.slug',
+            ],
+            $this->client->getContainer()
+        );
+    }
+
+    public function testEditSlugToTriggerRedirectionListener(): void
+    {
+        $this->authenticateAsAdmin($this->client, 'superadmin@en-marche-dev.fr', 'superadmin');
+
+        /** @var Article $article */
+        $article = $this->manager->getRepository(Article::class)->findOneBySlug('outre-mer');
+
+        $this->client->request(
+            Request::METHOD_GET,
+            sprintf('//'.$this->hosts['app'].'/articles/%s/%s', $article->getCategory()->getSlug(), $article->getSlug())
+        );
+
+        $this->assertStatusCode(Response::HTTP_OK, $this->client);
+
+        $this->client->request(
+            Request::METHOD_GET,
+            sprintf('//'.$this->hosts['amp'].'/articles/%s/%s', $article->getCategory()->getSlug(), $article->getSlug())
```

Comment:
Let's create a dedicated client for  `amp` subdomain. e.g.: 
`
$ampClient = $this->makeClient(false, ['HTTP_HOST' => $this->hosts['amp']]);`

This way we can get rid of every host declaration in urls, and network-like paths (starting  with `//`)